src/sender_queue/mod.rs

```diff
@@ -199,7 +199,11 @@ where
                     *e = epoch;
                 }
             }).or_insert(epoch);
-        self.process_new_epoch(sender_id, epoch)
+        if !self.remove_validator_if_old(sender_id) {
+            self.process_new_epoch(sender_id, epoch)
```

Comment:
I think we always need to call `process_new_epoch` first. Otherwise `remove_validator_if_old` might remove nonempty entries from `outgoing_queue`.

Reply:
Or we can remove here when the received epoch is greater than the last epoch.

Reply:
I'd find it a bit more natural and safer to do it the other way round. Waiting for an `EpochStarted(43)` from a validator that was removed in epoch 42 kind of relies on the fact that we currently don't detect our own removal, and pretend we can continue as an observer. (Not sure, though; I'm fine with keeping it this way for now.)